chrome/content/zotero/elements/zoteroSearch.js

```diff
@@ -0,0 +1,989 @@
+/*
+	***** BEGIN LICENSE BLOCK *****
+	
+	Copyright © 2022 Corporation for Digital Scholarship
+					 Vienna, Virginia, USA
+					 https://www.zotero.org
+	
+	This file is part of Zotero.
+	
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+	
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU Affero General Public License for more details.
+	
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero.  If not, see <http://www.gnu.org/licenses/>.
+	
+	***** END LICENSE BLOCK *****
+*/
+
+"use strict";
+
+{
+	var { Services } = ChromeUtils.import("resource://gre/modules/Services.jsm");
+	Services.scriptloader.loadSubScript("chrome://global/content/customElements.js", this);
+	Services.scriptloader.loadSubScript("chrome://zotero/content/elements/shadowAutocompleteInput.js", this);
+
+	let cssFiles = [
+		'chrome://global/skin/global.css',
+		'chrome://zotero-platform/content/zoteroSearch.css'
+	];
+
+	class SearchElementBase extends XULElement {
+		connectedCallback() {
+			var shadow = this.attachShadow({ mode: "open" });
+
+			for (let cssFile of cssFiles) {
+				var link = document.createElement("link");
+				link.rel = "stylesheet";
+				link.href = cssFile;
+				shadow.append(link);
+			}
+
+			if (this.content) {
+				let content = document.importNode(this.content, true);
+				shadow.appendChild(content);
+			}
+		}
+
+		disconnectedCallback() {
+			// Empty the DOM. We will rebuild if reconnected.
+			while (this.lastChild) {
+				this.removeChild(this.lastChild);
+			}
+		}
+	}
```

Comment:
Do we want to move this to a generic component (`XULElementBase`?) that we extend whenever we want automatic `this.content` or `this.stylesheets` importing?